hs_tools_resource/models.py

```diff
@@ -35,190 +33,112 @@ def metadata(self):
         md = ToolMetaData()
         return self._get_metadata(md)
 
-
 processor_for(ToolResource)(resource_processor)
 
 
 class RequestUrlBase(AbstractMetaDataElement):
-    term = 'Request Url Base'
-    value = models.CharField(null=True, max_length="500") # whatever the user gives us- format is "http://www.example.com/{resource-info}"
-
+    term = 'RequestUrlBase'
+    value = models.CharField(max_length=1024, null=True)
+    resShortID = models.CharField(max_length=128, default="UNKNOWN")
```

Comment:
I can't really provide a full example at this time, because it would take some research about handling your different meta data types. But, if you have any CoreMetaData object (or a subclass) you should be able to query for any BaseResource by object_id:

```
res = BaseResource.objects.get(object_id=metadata.id)
```

And you'd still need to find what type of resource that is and query for the correct model or proxy model.